diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt

```diff
@@ -102,7 +105,18 @@ class TrailingCommaRule(configRules: List<RulesConfig>) : DiktatRule(
             // we should write type of node in warning, to make it easier for user to find the parameter
             TRAILING_COMMA.warnAndFix(configRules, emitWarn, isFixMode, "after ${this.elementType}: ${this.text}", this.startOffset, this) {
                 val parent = this.treeParent
-                parent.addChild(LeafPsiElement(COMMA, ","), this.treeNext)
+
+                // In case, when we got VALUE_PARAMETER, it may contain comments, which follows the actual parameter and all of them are actually in the same node
+                // Ex: `class A(val a: Int, val b: Int  // comment)`
+                // `val b: Int  // comment` --> the whole expression is VALUE_PARAMETER
+                // So, in this case we must insert comma before the comment, in other cases we will insert it after current node
+                val comments = listOf(EOL_COMMENT, BLOCK_COMMENT, KDOC)
+                val firstCommentNodeOrNull = this.children().filter { it.elementType in comments }.firstOrNull()
```

Comment:
```suggestion
                val firstCommentNodeOrNull = this.children().firstOrNull { it.elementType in comments }
```
Also, these two calls can be merged

Reply:
Fixed